### PR TITLE
Refactor validity check for input files.

### PR DESCRIPTION
**Pull request by mburkart**
Accessing the files on dCache via xrootd consumes quite some time as a request is opened for each file and the list of files is quite long. The idea of the refactoring is to write a `.yaml` file for each dataset in the first run over a set of datasets. In the following runs this file can then be used to retrieve the information and the files only need to be opened when they are read.

### CODE REVIEW

File: booking.py [hunk 1]
```diff
@@ -121,7 +121,6 @@ def dataset_from_crownoutput(
     file_names,
     era,
     channel,
-    folder,
```
Comment (harrypuuter) [inline booking.py]: maybe we leave it in with a warning, since this will break frameworks using the function

**Review by harrypuuter — APPROVED** [review]
looks good ! 

how large is the speed improvement from this ?

File: booking.py (hunk 1 above)
Comment (mburkart) [reply booking.py]: Yes, we could consider this. These are two different functions though and you can anyway not switch between the inputs with this single change.

Comment (mburkart) [reply booking.py:124]: Yes, we could consider this. These are two different functions though and you can anyway not switch between the inputs with this single change.

**Pull request merged**